engine/lib/membrane_rtc_engine/engine.ex

```diff
@@ -611,6 +591,15 @@ defmodule Membrane.RTC.Engine do
   #   removed tracks. Also forwards custom media events.
   #
 
+  defp handle_endpoint_notification(:finished, endpoint_id, ctx, state) do
+    {{:present, endpoint}, actions, new_state} = handle_remove_endpoint(endpoint_id, ctx, state)
```

Comment:
We should handle `{:absent, ..}` too. Someone might try to remove endpoint at the same time endpoint tries to remove itself